integration-tests/kafka-sasl/src/test/java/io/quarkus/it/kafka/KafkaSASLTestResource.java

```diff
@@ -0,0 +1,124 @@
+package io.quarkus.it.kafka;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+import org.jboss.logging.Logger;
+import org.testcontainers.containers.FixedHostPortGenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.images.builder.Transferable;
+import org.testcontainers.utility.MountableFile;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+
+public class KafkaSASLTestResource implements QuarkusTestResourceLifecycleManager {
+
+    private final SaslStrimziKafkaContainer kafka = new SaslStrimziKafkaContainer()
+            .withCopyFileToContainer(MountableFile.forClasspathResource("server.properties"),
+                    "/opt/kafka/config/server.properties");
+
+    @Override
+    public Map<String, String> start() {
+        kafka.start();
+        // Used by the test
+        System.setProperty("bootstrap.servers", kafka.getBootstrapServers());
```

Comment:
Shouldn't the test be able to inject `kafka.bootstrap.servers` (which is added to the map further down) via `@ConfigProperty`? 🤔

Reply:
That's actually on purpose to distinguish the value retrieve in the test, and the once injected in the application. You can't inject the configuration value in the (native) test, so I use a system property.

Reply:
Ah, of course! Thanks.